lib/stm32/common/usart_common_all.c

```diff
@@ -94,6 +94,32 @@ void usart_set_baudrate(uint32_t usart, uint32_t baud)
 #endif
 }
 
+/*---------------------------------------------------------------------------*/
+/** @brief USART Get Baudrate.
+
+@param[in] usart unsigned 32 bit. USART block register address base @ref usart_reg_base
+@returns baud unsigned 32 bit. Baud rate specified in Hz.
+*/
+
+uint32_t usart_get_baudrate(uint32_t usart)
+{
+	uint32_t clock = rcc_get_usart_clk_freq(usart);
+	const uint32_t reg_brr = USART_BRR(usart);
+#ifdef USART_CR1_OVER8
+	if (USART_CR1(usart) & USART_CR1_OVER8) {
+		/* Need to shift BRR[2:0] up before using the simple formula of 2*clock/BRR (Q12.4) */
+		const uint16_t div_mantissa = reg_brr & USART_BRR_UPPER_MASK;
+		const uint16_t div_fractional = (reg_brr & USART_BRR_LOWER_MASK) << 1U;
+		const uint16_t div_over8 = div_mantissa + div_fractional;
+		return 2*clock / div_over8;
```

Comment:
Missing `U` suffix; please includes spaces around the operator and if you feel the need to show the multiplication must happen first, parens.

Reply:
Added. As long as USART kernel clock is below 2147 MHz, this multiplication should not overflow `uint32_t` (contrary to LPUART), and APB clocks are usually around 48-168 MHz (caveat dynamic kernel clocksource switching)